### PR TITLE
week 4 CFUs

### PR DESCRIPTION
**Pull request by vatrudell**

### CODE REVIEW

File: module-4/backend/week_four.md [hunk 1]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 
 Note: When you're done, submit a PR. 
 
-* What's the purpose of a project management tool?
+
+* What's the purpose of a project management tool?  
+    To organize the project into small managable parts
+    To keep track of what has already been done and what needs to be done 
+    To keep track of changes of the progress of each part including potentially the conversation around the progress
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: 👍 yay for code conversations!

File: module-4/backend/week_four.md [hunk 2]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+* What are the stages of a card when using Pivotal Tracker? 
+    icebox
+    current iteration
+    up next
+    done
+* What are the four types of cards available in Pivotal Tracker? 
+    started
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: These are stages/phases, the types are feature, bug, chore, and release.

File: module-4/backend/week_four.md [hunk 3]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+* What are the four types of cards available in Pivotal Tracker? 
+    started 
+    finished 
+    delivered 
+    accepted
 * What's your favorite project management tool?
-* Why do we create staging environments?
+    Waffle :3
+* Why do we create staging environments? 
+    so that we can work with the production environment (different api keys + variable names + databases) in a way that does
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: 👍

File: module-4/backend/week_four.md [hunk 4]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+    so that we can work with the production environment (different api keys + variable names + databases) in a way that does      
+    not effect production in anyway. 
 * What are the characteristics of a good README (in your opinion)?
-* What's one main improvement you're going to make to your code regarding accessibility issues?
-* What are some basic security concerns to be aware of when building applications?
-* Why is continuous integration helpful/important?
+    - A good explaination of how you use the tool 
+    - Listing any endpoints and how you interact with them 
+    - Making it begginner friendly 
+    - Explains how the application was build, what tools were used
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: ...and how to get a local environment set up!

File: module-4/backend/week_four.md [hunk 5]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+* What's one main improvement you're going to make to your code regarding accessibility issues? 
+    - Make sure you can read through the whole of the available pages and the text to voice program pick up 
+      all the intended text 
+    - Making the webpages calm and minimal as possible
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: 👍

File: module-4/backend/week_four.md [hunk 6]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+            
+            - don't store passwords in plain text
+            - at registration that users make passwords strong
+            - using strong params when using incoming data
+            - sanitizing the incoming data 
+            - keeping certain information secure -> whether some pages do or do not exist, if the login was incorrect 
+              revealing which ones were wrong
+          
+* Why is continuous integration helpful/important? 
+    contineous integration is helpful so that you can test the application in a different environment
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: and it promotes the agile philosophy. Deploy often!

File: module-4/backend/week_four.md [hunk 7]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
 * What are some continuous integration tools?
+    Travis CI
 
 #### Review  
 
 * What are some characteristics of "good" git workflow?
+    small commit messages
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: I'm not sure this is always the case. The first line should always be a quick summary, but it is totally fine (and sometimes necessary) to add details in the body.

File: module-4/backend/week_four.md [hunk 8]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+    commiting often
+    not working on master
+    having a clear flow of branches merging back into master
 * What are the four fundamental concepts of object oriented programming?
+    Encapsulation - puting things into small functional parts
+    Abstraction - keeping the current file/method as small as possible and taking out anything that doesn't do what the name 
+          says it does 
+    Inheritance - files can inherit from other files. things don't need to be in the same place
+    Writing it once, using it over and over
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: Polymorphism!

File: module-4/backend/week_four.md [hunk 9]
```diff
@@ -4,17 +4,65 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
 * What's a module in Ruby and what's the difference between a class and a module?
+    classes inherit from modules. Modules are the blueprint of how something might look and act but class inherit it and use
```
Comment (tmikeschu) [inline module-4/backend/week_four.md]: I would do some more research on this. Classes tend to `extend` or `include` modules. Modules are stateless, and typically provide shared functions that many classes can use.

Comment (tmikeschu) [inline module-4/backend/week_four.md]: @case-eee @neight-allen am I way off here?

Comment (case-eee) [inline module-4/backend/week_four.md]: @tmikeschu i agree 👍

**Pull request closed without merging**